wgpu-core/src/instance.rs

```diff
@@ -286,76 +203,9 @@ impl<F: IdentityFilter<AdapterId>> Global<F> {
             PowerPreference::HighPerformance => discrete.or(other).or(integrated).or(virt),
         };
 
-        let mut token = Token::root();
-        let mut selected = preferred_gpu.unwrap_or(0);
-        #[cfg(any(
-            not(any(target_os = "ios", target_os = "macos")),
-            feature = "gfx-backend-vulkan"
-        ))]
-        {
-            if selected < adapters_vk.len() {
-                let adapter = Adapter {
-                    raw: adapters_vk.swap_remove(selected),
-                };
-                log::info!("Adapter Vulkan {:?}", adapter.raw.info);
-                let id = backend::Vulkan::hub(self).adapters.register_identity(
-                    id_vulkan.unwrap(),
-                    adapter,
-                    &mut token,
-                );
-                return Some(id);
-            }
-            selected -= adapters_vk.len();
-        }
-        #[cfg(any(target_os = "ios", target_os = "macos"))]
-        {
-            if selected < adapters_mtl.len() {
-                let adapter = Adapter {
-                    raw: adapters_mtl.swap_remove(selected),
-                };
-                log::info!("Adapter Metal {:?}", adapter.raw.info);
-                let id = backend::Metal::hub(self).adapters.register_identity(
-                    id_metal.unwrap(),
-                    adapter,
-                    &mut token,
-                );
-                return Some(id);
-            }
-            selected -= adapters_mtl.len();
-        }
-        #[cfg(windows)]
-        {
-            if selected < adapters_dx12.len() {
-                let adapter = Adapter {
-                    raw: adapters_dx12.swap_remove(selected),
-                };
-                log::info!("Adapter Dx12 {:?}", adapter.raw.info);
-                let id = backend::Dx12::hub(self).adapters.register_identity(
-                    id_dx12.unwrap(),
-                    adapter,
-                    &mut token,
-                );
-                return Some(id);
-            }
-            selected -= adapters_dx12.len();
-            if selected < adapters_dx11.len() {
-                let adapter = Adapter {
-                    raw: adapters_dx11.swap_remove(selected),
-                };
-                log::info!("Adapter Dx11 {:?}", adapter.raw.info);
-                let id = backend::Dx11::hub(self).adapters.register_identity(
-                    id_dx11.unwrap(),
-                    adapter,
-                    &mut token,
-                );
-                return Some(id);
-            }
-            selected -= adapters_dx11.len();
-        }
+        let selected = preferred_gpu.unwrap();
```

Comment:
This should not be able to fail, we already exit the function early if there is no adapter available and as far as I can see there is no code path that could prevent this part of the code to not select a GPU at the end.